### PR TITLE
feat(#2266): Update jcabi-xml library in order to use XPath 2.0

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
Update `jcabi-xml` library in order to use `XPath 2.0` and enable the `discoversWithSeveralObjectsWithDifferentVersions` test.

Closes: #2266

---

### Focus of this PR:
Update dependencies and make code improvements.

### Detailed summary:
- Update `jcabi-xml` version from `0.27.2` to `0.28.0` in `pom.xml`
- Change scope of `Saxon-HE` dependency from `runtime` to `compile` in `eo-maven-plugin/pom.xml`
- Add import statement for `com.jcabi.xml.SaxonDocument` in `DiscoverMojo.java`
- Add import statements for `org.cactoos.text.TextOf` and `org.cactoos.text.UncheckedText` in `DiscoverMojo.java`
- Use `SaxonDocument` constructor with `String` parameter instead of `XMLDocument` in `DiscoverMojo.java`
- Add import statement for `org.eolang.maven.hash.CommitHash` in `DiscoverMojoTest.java`
- Enable `discoversWithSeveralObjectsWithDifferentVersions` test in `DiscoverMojoTest.java`
- Change expected values in `discoversWithSeveralObjectsWithDifferentVersions` test in `DiscoverMojoTest.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
@maxonfjvipon Could you have a look, please?

File: eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java [hunk 1]
```diff
@@ -149,12 +142,13 @@ void discoversWithVersions(@TempDir final Path tmp) throws IOException {
     }
 
     @Test
-    @Disabled
     void discoversWithSeveralObjectsWithDifferentVersions(
         @TempDir final Path tmp
     ) throws IOException {
+        final CommitHashesMap.Fake hashes = new CommitHashesMap.Fake();
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java:148]: @volodya-lombrozo What is the point of specifying the exact type here - `CommitHashesMap.Fake`? Maybe it's better to use just `CommitHashesMap`?

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo see above

File: eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java (hunk 1 above)
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java]: @maxonfjvipon We can't use `CommitHashesMap` here, but we can use `Map<String, CommitHash>`. Thanks.

File: eo-maven-plugin/src/main/java/org/eolang/maven/DiscoverMojo.java [hunk 2]
```diff
@@ -93,14 +96,21 @@ public void exec() throws FileNotFoundException {
...
-        final Collection<String> names = DiscoverMojo.names(xml, this.xpath(false));
+        final XML saxon = new SaxonDocument(new UncheckedText(new TextOf(file)).asString());
+        final Collection<String> names = DiscoverMojo.names(saxon, this.xpath(false));
         if (this.withVersions) {
             names.addAll(
-                DiscoverMojo.names(xml, this.xpath(true))
+                DiscoverMojo.names(saxon, this.xpath(true))
             );
         }
         if (!xml.nodes("//o[@vararg]").isEmpty()) {
```
[the remark this answers is not in the document]
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/main/java/org/eolang/maven/DiscoverMojo.java:116]: @maxonfjvipon `saxon` implements only `xpath` function, all the other functions are not implemented yet. So, we can't replace it.

[the remark this answers is not in the document]
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/main/java/org/eolang/maven/DiscoverMojo.java]: @maxonfjvipon done

**Comment by volodya-lombrozo** [comment]
@maxonfjvipon Have a look one more time please.

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you merge that changes, please?

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**